### ISSUE TITLE
warning adding a new addressbook 

### ISSUE THREAD

**Issue by 532910** [blind-coder/rcmcarddav#273]
Every time I configure a new addressbook (via web interface or preset) I got:

```
PHP Warning:  dns_get_record(): An unexpected server failure occurred. in /srv/www/roundcube/plugins/carddav-3.0.3/carddav_discovery.php on line 252
PHP Warning:  Invalid argument supplied for foreach() in /srv/www/roundcube/plugins/carddav-3.0.3/carddav_discovery.php on line 253
```

But all works fine.

**Comment by troglodyne** [comment]
Error can be ignored. It is looking for a DNS_TXT record, but it doesn't exist. Doesn't normally hurt things, as the code just falls back to assuming the server address you gave is legit.

I just patched the line in question to use @ to silence warn from dns_get_record then added an empty() check before the foreach to skip the loop in that case

**Comment by mstilkerich** [comment]
I don't get these warnings with PHP 7.2. Are you using an older version of PHP? It is a quite common case that these lookups will not return a TXT record.

**Comment by 532910** [comment]
7.3 from debian stable

**Comment by mstilkerich** [comment]
I cannot reproduce it on my end. I now also tried php 7.4 (I have ubuntu 18.04 and 20.04 with php 7.2 and 7.4).

Does this script display the warning for you:

```
<?php
error_reporting(32767);
$dnsresults = dns_get_record("_carddavs._tcp.gmail.com", DNS_TXT);
var_dump($dnsresults);
```

It's basically exactly what the plugin does, gmail does not have a TXT record for the carddavs service.

**Comment by 532910** [comment]
```
[sergio@roundcube:/tmp]% php t.php 
array(0) {
}
[sergio@roundcube:/tmp]%
```

**Comment by mstilkerich** [comment]
Ok, I looked at the current master branch when I should have looked at the 3.0.3 tag. The warning occurred because the discovery code tried lookups for an empty record name. This was fixed with pull request commit e6138e23a31ec099306383c42a9a02231d8ef2b1, which in turn broke discovery for services that do not have DNS SRV records set up. Anyway, the v4 branch has a fixed discovery, so I'm closing this issue. Please use the v4 branch if you are affected by this issue.
